src/scipp/reduction.py

```diff
@@ -0,0 +1,63 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+# @author Simon Heybrock
+import uuid
+from typing import Union, List, Tuple
+from .typing import VariableLike
+from .core import concat, reduction
+
+
+def _make_method(f):
+    def method(self) -> VariableLike:
+        return f(self._obj, self._dim)
+
+    return method
```

Comment:
This results in a method with a complicated name and no help message:
```.py
>>> r = sc.reduce([sc.array(dims=['x'], values=[1])])
>>> r.all
<bound method _make_method.<locals>.method of <scipp.reduction.Reducer object at 0x7ffa5d16d550>>
>>> help(r.all)
Help on method method in module scipp.reduction:

method() -> Union[scipp._scipp.core.Variable, scipp._scipp.core.DataArray, scipp._scipp.core.Dataset] method of scipp.reduction.Reducer instance
```
You could use something similar to
https://github.com/scipp/scipp/blob/f323f55c04a2a94055274530992f72bd54f51cf1/src/scipp/_binding.py#L35
to propagate the function name, make it look like it was defined in `Reducer`, and give it a docstring (though you would need a new docstring, the ones of the free functions don't really work here.)